pkg/services/signingkeys/signingkeysimpl/service.go

```diff
@@ -30,7 +31,10 @@ func ProvideEmbeddedSigningKeysService() (*Service, error) {
 		return nil, signingkeys.ErrKeyGenerationFailed.Errorf("Error generating private key: %v", err)
 	}
 
-	if err := s.AddPrivateKey(serverPrivateKeyID, privateKey); err != nil {
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer cancel()
+	expiry := time.Now().Add(30 * 24 * time.Hour)
+	if err := s.AddPrivateKey(ctx, signingkeys.ServerPrivateKeyID, privateKey, jose.ES256, &expiry, false); err != nil {
```

Comment:
@chagrafana @jmatosgrafana 

If you have views on the default rotation of signing keys (set to 30 days currently) and the algo, please comment here

Reply:
Algortithm is fine (ECDSA 256 bit) and 30 days retention is OK as long as we trust the encrypted storage in database.
I left a comment related to that [here](https://github.com/grafana/grafana/pull/75487/files#r1338660797).
It's all about the key management for the encryption of this private key